src/content/docs/ar/core-concepts/astro-components.mdx

```diff
@@ -0,0 +1,298 @@
+
+<!-- Mix HTML with JavaScript expressions, similar to JSX: -->
+<ul>
+  {myFavoritePokemon.map((data) => <li>{data.name}</li>)}
+</ul>
+
+<!-- Use a template directive to build class names from multiple strings or even objects! -->
+<p class:list={["add", "dynamic", {classNames: true}]} />
+```
+
+## التصميم القائم على المكوّنات
+
+المكوّنات مصممة لتكون قابلة **لإعادة الاستخدام** و**التركيب**. يمكنك استخدام مكوّنات داخل مكوّنات أخرى لبناء واجهات مستخدم متقدمة بشكل أكبر وأكثر تطورًا. على سبيل المثال، يمكن استخدام مكوّن "زر" (Button) لإنشاء مكوّن "مجموعة أزرار" (ButtonGroup):
+
+
+```astro title="src/components/ButtonGroup.astro"
+---
+import Button from './Button.astro';
+---
+<div>
+  <Button title="Button 1" />
+  <Button title="Button 2" />
+  <Button title="Button 3" />
+</div>
+```
+
+
+## خاصيّات المكوّن
+
+يمكن لمكوّن أسترو أن يعرّف ويستقبل الخاصيّات "props". هذه الخاصيّات  تصبح متاحة بعدها لقالب المكوّن لتصيير HTML. الخاصيّات  متاحة في `Astro.props` .بشكل عام في البرنامج النصّي  للـfrontmatter  الخاص بك
+
+هنا مثال لمكوّن يستقبل خاصيّة `greeting` وخاصيّة `name`. لاحظ أن الخاصيّات  التي سيتم استقبالها مفككة من object `Astro.props` العام.
+
+```astro "Astro.props"
+---
+// src/components/GreetingHeadline.astro
+// Usage: <GreetingHeadline greeting="Howdy" name="Partner" />
+const { greeting, name } = Astro.props;
+---
+<h2>{greeting}, {name}!</h2>
+```
+عند استيراد هذا المكوّن وتصييره في مكونات أو تخطيطات أو صفحات أسترو أخرى، يمكن تمرير هذه الخاصيّات "props" كسمات "attributes":
+
+```astro /(\w+)=\S+/
+---
+// src/components/GreetingCard.astro
+import GreetingHeadline from './GreetingHeadline.astro';
+const name = "Astro"
+---
+<h1>Greeting Card</h1>
+<GreetingHeadline greeting="Hi" name={name} />
+<p>I hope you have a wonderful day!</p>
+```
+
+بالإمكان أيضًا تعريف الخاصيّات الخاصة بك مع TypeScript باستخدام واجهة `Props`. سيقوم أسترو تلقائيًا باكتشاف واجهة `Props` في الـ "frontmatter" الخاص بك وإعطاء تحذيرات أو أخطاء إن وجد. يمكن أيضًا إعطاء هذه الخاصيات قيمًا افتراضية عند استخراجها من `Astro.props`.
+
+```astro ins={3-6}
+---
+// src/components/GreetingHeadline.astro
+interface Props {
+  name: string;
+  greeting?: string;
+}
+
+const { greeting = "Hello", name } = Astro.props;
+---
+<h2>{greeting}, {name}!</h2>
+```
+يمكن إعطاء خاصيّات  المكوّنات قيمًا إفتراضية عندما لا يتم توفير أي منها.
+
+```astro ins="= \"Hello\"" ins="= \"Astronaut\""
+---
+// src/components/GreetingHeadline.astro
+const { greeting = "Hello", name = "Astronaut" } = Astro.props;
+---
+<h2>{greeting}, {name}!</h2>
+```
+
+## المداخل
+
+`<slot />` هو عنصر نائب لمحتوى HTML الخارجي، مما يسمح لك بحقن (أو "ادخال") عناصر من ملفّات أخرى إلى قالب مكوّنك.
+
+بشكل افتراضي، سيتم تصيير كل العناصر الداخلية الممرّرة لمكوّن في `<slot />` الخاص به.
+
+:::note
+بعكس _الخاصيات _، التي هي سمات "attributes" يتم تمريرها إلى مكوّن أسترو والمتاحة للإستخدام في المكوّن الخاص بك باستخدام `Astro.props`، _المداخل "slots"_ تُصيّر عناصر HTML الداخليّة حيث كتبت.
```

Comment:
<img width="851" alt="Screenshot 2023-07-29 at 2 08 51 AM" src="https://github.com/withastro/docs/assets/118110192/68924f23-9adf-42f7-9213-f52144c1954b">

this has not been resolved